exercises/list-ops/example.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+function List() {
+  return {
+    append: function (arrOne, arrTwo) {
+      var appended = [];
+
+      for (var i = 0; i < arrOne.length; i++) {
+        appended.push(arrOne[i]);
+      }
+
+      for (var j = 0; j < arrTwo.length; j++) {
+        appended.push(arrTwo[j]);
+      }
+
+      return appended;
+    },
+
+    concat: function (arr) {
+      var concatenated = [];
+
+      for (var i = 0; i < arr.length; i++) {
+        concatenated = this.append(concatenated, arr[i]);
+      }
+
+      return concatenated;
+    },
+
+    cons: function (x, arr) {
+      return this.append([x], arr);
+    },
+
+    foldl: function (func, start, arr) {
+      var acc = start;
+
+      for (var i = 0; i < arr.length; i++) {
+        acc = func(arr[i], acc);
+      }
+
+      return acc;
+    },
+
+    foldr: function (func, start, arr) {
+      var acc = start;
+
+      for (var i = arr.length - 1; i >= 0; i--) {
+        acc = func(arr[i], acc);
+      }
+
+      return acc;
+    },
+
+    length: function (arr) {
+      var increment = (x, acc) => acc + 1;
+      return this.foldl(increment, 0, arr);
+    },
+
+    reverse: function (arr) {
+      return this.foldl(this.cons.bind(this), [], arr);
```

Comment:
fold is magic ✨
my thought process: cons is the identity function for foldr, so by switching to foldl, we get the same list back, but reversed!